contracts/governance/Vesting/fouryear/FourYearVestingFactory.sol

```diff
@@ -10,31 +10,27 @@ import "./IFourYearVestingFactory.sol";
  * of the same contract and keep track of them easier.
  * */
 contract FourYearVestingFactory is IFourYearVestingFactory, Ownable {
-	address public fourYearVestingLogic;
-
-	constructor(address _fourYearVestingLogic) public {
-		require(_fourYearVestingLogic != address(0), "invalid four year vesting logic address");
-		fourYearVestingLogic = _fourYearVestingLogic;
-	}
-
 	/**
 	 * @notice Deploys four year vesting contract.
 	 * @param _SOV the address of SOV token.
 	 * @param _staking The address of staking contract.
 	 * @param _tokenOwner The owner of the tokens.
 	 * @param _feeSharing The address of fee sharing contract.
-	 * @param _vestingOwner The address of an owner of vesting contract.
+	 * @param _vestingOwnerMultisig The address of an owner of vesting contract.
+	 * @dev _vestingOwnerMultisig should ALWAYS be multisig.
+	 * @param _fourYearVestingLogic The implementation contract.
 	 * @return The four year vesting contract address.
 	 * */
 	function deployFourYearVesting(
 		address _SOV,
 		address _staking,
 		address _tokenOwner,
 		address _feeSharing,
-		address _vestingOwner
+		address _vestingOwnerMultisig,
+		address _fourYearVestingLogic
 	) external onlyOwner returns (address) {
-		address fourYearVesting = address(new FourYearVesting(fourYearVestingLogic, _SOV, _staking, _tokenOwner, _feeSharing));
-		Ownable(fourYearVesting).transferOwnership(_vestingOwner);
+		address fourYearVesting = address(new FourYearVesting(_fourYearVestingLogic, _SOV, _staking, _tokenOwner, _feeSharing));
+		Ownable(fourYearVesting).transferOwnership(_vestingOwnerMultisig);
```

Comment:
what is the reason for this contract to exist? we can deploy the `FourYearVesting` contract directly. setting ownership transferring to the contract would be enough. also, for the testnet it is sometimes suboptimal to force transferring ownership to the multisig.

Reply:
Also, testnet testing should not be a problem as the vesting contracts will be created for known token owner(my accounts)

Reply:
As you can see factory includes logic contract that can be relatively big, so using factory we separate bytecode of logic contact from bytecode of other contracts (e.g. VestingRegistry), otherwise we need to include bytecode of logic contract to VestingRegistry and can easily exceed 24 kb.